### DISCUSSION TITLE
Binary vs data classification exponentially slower on EC2

### DISCUSSION THREAD

**Discussion by luisldq** [pyinstaller/pyinstaller#8794] [Help] [answered]
Has anyone else experienced extreme slowness running pyinstaller builds on EC2, to the point of never finishing? On my cheap 16gb Ryzen 5 machine the build takes about an hour.

On EC2 however the build hangs at Binary vs Data Classification, when log-level=TRACE I can see it takes minutes to classify tiny DLL files, given there are hundreds of thousands of files, it never finishes even after 24 hours.

I've tried M, T, and C instances with 16gb ram and 4 cores, all behaving the same.

I see no IO activity in Task Manager, RAM is 1.5gb with plenty to spare, only CPU activity, but with plenty to spare.

Probably an EC2 user error, but wondering if anyone else has had the same problem.

**Comment by rokm** [comment, marked as answer]
>  I can see it takes minutes to classify tiny DLL files

Sounds like https://github.com/pyinstaller/pyinstaller/issues/8762.

**Comment by luisldq** [comment]
Thank you @rokm ! I downgraded pefile to 2023.2.7 and all is well now.
